state/remote/artifactory.go

```diff
@@ -6,27 +6,50 @@ import (
 	"os"
 	"strings"
 
-	artifactory "github.com/lusis/go-artifactory/src/artifactory.v401"
+	artifactory "github.com/lusis/go-artifactory/artifactory.v54"
 )
 
+// ARTIF_TFSTATE_NAME is the constant for the terraform state file stored in artifactory
 const ARTIF_TFSTATE_NAME = "terraform.tfstate"
```

Comment:
Can I suggest not to force the filename here to be able to use a free form as it is possible with the s3 backend ? Either by including it in `subpath` or by adding an option like `statefile` that would override the default `terraform.tfstate` ?